plugins/inputs/aliyuncms/discovery.go

```diff
@@ -21,39 +22,15 @@ import (
 	"github.com/pkg/errors"
 )
 
-// https://www.alibabacloud.com/help/doc-detail/40654.htm?gclid=Cj0KCQjw4dr0BRCxARIsAKUNjWTAMfyVUn_Y3OevFBV3CMaazrhq0URHsgE7c0m0SeMQRKlhlsJGgIEaAviyEALw_wcB
-var aliyunRegionList = []string{
-	"cn-qingdao",
-	"cn-beijing",
-	"cn-zhangjiakou",
-	"cn-huhehaote",
-	"cn-hangzhou",
-	"cn-shanghai",
-	"cn-shenzhen",
-	"cn-heyuan",
-	"cn-chengdu",
-	"cn-hongkong",
-	"ap-southeast-1",
-	"ap-southeast-2",
-	"ap-southeast-3",
-	"ap-southeast-5",
-	"ap-south-1",
-	"ap-northeast-1",
-	"us-west-1",
-	"us-east-1",
-	"eu-central-1",
-	"eu-west-1",
-	"me-east-1",
-}
-
 type discoveryRequest interface {
 }
 
 type aliyunSdkClient interface {
 	ProcessCommonRequest(req *requests.CommonRequest) (response *responses.CommonResponse, err error)
 }
 
-type discoveryTool struct {
+// DiscoveryTool is a object that provides discovery feature
+type DiscoveryTool struct {
```

Comment:
Why do you need to export it?

Reply:
Well, actually the discovery tool can be used as a separate object that provides discovery feature,  and hypothetically can  be pushed to a separate (from the plugin) package. 
If this smth. that is a kind of way too early, then we can revert  back to non exported type and channel.
However please not that there is also some methods of `DiscoveryTool` object that are exported (again keeping in mind this possibility).
Please do let me know your opinion, and I can push updated to this PR.

Reply:
I prefer to not change it in this PR as it's kind of a scope creeping. Furthermore, if it will become a common thing to be used by other packages we should move it to `plugins/common`.

Reply:
Ok, noted, I will switch it to non-exported.